test/kyberNetwork.js

```diff
@@ -3548,207 +3548,207 @@ contract('KyberNetwork', function(accounts) {
+                    let reportedBalance = await token.balanceOf(reserve4.address);
+                    assert.equal(reportedBalance.valueOf(), reserve4TokenBalance[tokenInd].valueOf(), "bad token balance on reserve");
+                } catch (e) {
+                    console.log("oooops " + e);
+                    throw e;
+                }
+            }
+            console.log("Average gas used for " + numTrades + " eth to token trades: " + gasUsed / numTrades);
+        });
+
+        it("should only enable new reserve. perform buy and check: balances changed as expected.", async function () {
+            let tokenInd = 1;
             let token = tokens[tokenInd]; //choose some token
-            let amountTwei = 100 * step + 20;
+            let amountWei = 330 * 1;
             let user = accounts[9];
             try {
                 //verify base rate
-                let rate = await network.getExpectedRate(tokenAdd[tokenInd], ethAddress, amountTwei);
-                let expected = calculateRateAmountNewConversionRate(false, tokenInd, amountTwei);
-                let expectedRate = expected[0].valueOf();
-                let expectedAmountWei = expected[1].valueOf();
-                expectedRate = calcCombinedRate(amountTwei, expectedRate, precisionUnits, tokenDecimals[tokenInd], 18, expectedAmountWei);
-                //check correct rate calculated
-                assert.equal(rate[0].valueOf(), expectedRate.valueOf(), "unexpected rate.");
+                let buyRate = await network.getExpectedRate(ethAddress, tokenAdd[tokenInd], amountWei);
+                let expected = calculateRateAmountNewConversionRate(true, tokenInd, amountWei);
+                let expectedRate = expected[0];
+                let expectedTweiAmount = expected[1];
+                let oldUserTokenBal = await token.balanceOf(user);
 
-                await token.transfer(network.address, amountTwei);
+                expectedRate = calcCombinedRate(amountWei, precisionUnits, expectedRate, 18, tokenDecimals[tokenInd], expectedTweiAmount);
+
+                //check correct rate calculated
+                assert.equal(buyRate[0].valueOf(), expectedRate.valueOf(), "unexpected rate.");
 
                 //perform trade
-                let txData = await network.tradeWithHint(user, tokenAdd[tokenInd], amountTwei, ethAddress, user, 500000,
-                                rate[1].valueOf(), walletId, 0, {from:networkProxy});
-                if (step < 2) {
-                    console.log("Transaction gas used token to eth for " + (step + 1) + " sell steps: " + txData.receipt.gasUsed);
-                }
-                //check lower ether balance on reserve
-                expectedReserve4BalanceWei = expectedReserve4BalanceWei.sub(expectedAmountWei);
+                let txData = await network.tradeWithHint(user, ethAddress, amountWei, tokenAdd[tokenInd], user, 50000,
```

Comment:
this is part of the changes?

Reply:
Nope, shifting stuff into a separate 'describe' section causes the difference check to go a bit wonky.